packages/documentation/copy/en/developer/ROS Migration Guide.md

```diff
@@ -0,0 +1,427 @@
+  system as `colcon` (and `ament`), and thus can nicely interoperate with
+  CMake-based ROS packages (which include most C/C++ packages). To add an existing ROS
+  package or node as a build dependency, you could make use of the
+  [`cmake-include`](/docs/handbook/target-declaration#cmake) target parameter.
+  For example, to use `rclcpp`, you could do the following:
+
+  ```lf-c
+  target CCpp {
+      cmake-include: ["rclcpp-support.cmake",]
+  };
+  ```
+
+  And write the `rclcpp-support.cmake` as follows:
+
+  ```lf-c
+  find_package(ament_cmake REQUIRED)
+  find_package(rclcpp REQUIRED)
+  ament_target_dependencies(${LF_MAIN_TARGET} rclcpp)
+  ```
+
+  Note the use of `ament_target_dependencies`, which is not a built-in CMake
+  method. `ament_target_dependencies` is a CMake method that is provided as part
+  of the
+  [`ament_cmake`](https://docs.ros.org/en/humble/How-To-Guides/Ament-CMake-Documentation.html#ament-cmake-user-documentation)
+  package and is more or less a shortcut for including and linking target
+  dependencies (including any relevant directories that may contain header
+  files).
+
+  Also note that ROS setup files should be properly [sourced](https://docs.ros.org/en/humble/Tutorials/Configuring-ROS2-Environment.html#source-the-setup-files) in order for `rclcpp` to be findable.
+
+- **Built-in Support for ROS 2 Message Types:** The CCpp target has built-in
+  support for ROS 2 message (and service) types. Messages are still transported
+  on (deterministic) Lingua Franca connections, but reactor ports can have ROS
+  2 message (or service) types. In federated execution, rclcpp's serializer is
+  transparently used to serialize messages.
+
+  For example, you could write a
+  simple Lingua Franca program where a `Sender` reactor sends a
+  `std_msgs::msg::Int32` to a `Receiver` reactor as follows:
+
+  ```lf-c
+  target CCpp {
+      cmake-include: ["ros-support.cmake",]
+  };
+  preamble {=
+      #include "std_msgs/msg/int32.hpp"
+  =}
+  reactor Sender {
+      output out:std_msgs::msg::Int32;
+      reaction (startup) -> out {=
+          std_msgs::msg::Int32 ros_message;
+          ros_message.data = 1;
+          lf_set(out, ros_message);
+      =}
+  }
+  reactor Receiver {
+    input in:std_msgs::msg::Int32;
+    reaction (in) {=
+        // Print the ROS2 message data
+        lf_print("Received: %d", in->value.data);
+    =}
+  }
+  federated reactor { // Also works with main
+      sender = new Sender();
+      receiver = new Receiver();
+      sender.out -> receiver.in serializer "ros2";
+  }
+  ```
+
+  <img alt="Lingua Franca diagram" src="../../../../../img/ros_migration/Sender-Receiver.svg" width="350"/>
+
+  The `ros-support.cmake` simply contains:
+
+  ```lf-c
+  find_package(std_msgs REQUIRED)
+  ament_target_dependencies(${LF_MAIN_TARGET} std_msgs)
+  ```
+
+  Note the use of `serializer "ros2"` on the connection between the sender and
```

Comment:
As a side note, I don't think we have any documentation for serializers on the website, do we?